src/components/NewWorkspaceModal.js

```diff
@@ -65,33 +71,40 @@ export default class NewWorkspaceModal extends Component {
   }
 
   async create() {
-    const { onCreate } = this.props
+    const { onCreate, cloneWorkspace } = this.props
     const { namespace, name, description, groups } = this.state
     try {
-      this.setState({ createError: undefined, creating: true })
-      await Rawls.createWorkspace({
+      this.setState({ createError: undefined, busy: true })
+      const body = {
         namespace,
         name,
         authorizationDomain: _.map(v => ({ membersGroupName: v }), groups),
         attributes: { description }
-      })
+      }
+      await (cloneWorkspace ?
+        Rawls.workspace(cloneWorkspace.workspace.namespace, cloneWorkspace.workspace.name).clone(body) :
+        Rawls.createWorkspace(body))
       onCreate({ namespace, name })
     } catch (error) {
-      this.setState({ createError: JSON.parse(error).message, creating: false })
+      this.setState({ createError: JSON.parse(error).message, busy: false })
     }
   }
 
   render() {
-    const { onDismiss } = this.props
-    const { namespace, name, billingProjects, allGroups, groups, description, nameModified, creating, createError } = this.state
+    const { onDismiss, cloneWorkspace } = this.props
+    const { namespace, name, billingProjects, allGroups, groups, description, nameModified, busy, createError } = this.state
+    const existingGroups = _.map(
```

Comment:
Is there no way to express this more simply?

Reply:
Not that I can think of... either of `cloneWorkspace` or `authorizationDomain` could be undefined, and `_.map` returns `[]` in those cases, which is basically what we want.